### DISCUSSION TITLE
How to know when rendering complete

### DISCUSSION THREAD

**Discussion by rebmaso** [pelicanmapping/osgearth#2623] [Support Q&A]
Hello,

Is there any way to tell If OsgEarth is done rendering to a sufficient level of detail, after setting a new camera view (e.g. jumping to "home" location in the viewer app)

I have tried using openscenegraph's [DrawCallback](https://stackoverflow.com/questions/17381229/how-can-i-capture-an-image-for-post-processing-in-osg) method but I don't think it does what I want.

Thanks a lot! 🙏

**Comment by gwaldron** [comment]
The general answer is "no" since osgEarth loads levels of detail incrementally and has no consistent way of knowing whether more, higher-resolution data will be available from the source until it asks.

osgEarth loads tile data in background jobs. So one thing you can try is to query the total number of jobs that are running. Once this number drops to zero and stays there your data is done loading.
```c++
auto num_running_jobs = jobs::get_metrics()->total();
```
